puppet/modules/slave/manifests/postgresql.pp

```diff
@@ -16,6 +16,15 @@
           ensure => 'present',
           before => Class['postgresql::globals'],
         }
+        yumrepo { 'foreman-infra-el7':
+          baseurl =>  'https://yum.theforeman.org/infra/el7/',
+          enabled => true,
+          require => Class['postgresql::globals'],
+        }
```

Comment:
This repository can be setup at any point.

Reply:
Do you mean that the infra repo doesn't need a `require`? Or that it doesn't need to be in the `if $facts['os']['name'] == 'CentOS'`?

Reply:
The repo doesn't need a require. As in, it can technically be laid down at any point. The package installation is what matters order of operations wise.

Reply:
There won't be a case where puppet tries to install the evr package before the infra repo is enabled?

Reply:
Oh yes, there could be, you can add a `} ->` here.